lib/orders/use_case/add_beverage_to_order.rb

```diff
@@ -0,0 +1,34 @@
+module Orders
+  module UseCase
+    class AddBeverageToOrder
+      Request = Struct.new(:order_id, :beverage)
+
+      attr_reader :gateway
+
+      def initialize(gateway: Gateway::Order.new)
```

Comment:
I'd avoid constructing gateways here, I've seen it done but it's better to put all constructional logic in one location. See use case factory.